### PR TITLE
add custom commands into client and crud test

### PR DESCRIPTION
**Pull request by ravlio**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -1202,6 +1202,26 @@ export class StreamChat {
 		return this.get(this.baseURL + `/channeltypes`);
 	}
 
+	createCommand(data) {
```
Comment (vishalnarkhede) [inline src/client.js]: We need to update typescript [file](https://github.com/GetStream/stream-chat-js/blob/master/types/stream-chat/index.d.ts) for these changes. If you familiar with typescript, please do it ... or let me know. I will update it once this PR is approved.

File: test/http_mock/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const app = express();
+app.use(express.json());
+module.exports = {
+	start() {
```
Comment (vishalnarkhede) [inline test/http_mock/index.js]: What do we need this server for exactly? (just want to know)

Comment (ravlio) [reply test/http_mock/index.js]: I use it for the actionHandler stub (our new feature for custom commands). client sendMessage -> chat server actionHandler->customer Server->chat server. So I'm mocking *customer Server*

Comment (vishalnarkhede) [reply test/http_mock/index.js]: Ok can you please add that description as comment somewhere in file? Also we should stop the server after corresponding tests are executed. I don't see `stop()` getting called anywhere (unless I am missing something)

**Pull request closed without merging**